`ot_earlgrey`: Stop disabling lc_ctrl state for Ibex

AlexJones0 opened the pull request:
See relevant comment here: https://github.com/lowRISC/qemu/pull/269#issuecomment-3469650725

I believe that this comment & property are now redundant because the LC state should be sufficiently emulated within the Earlgrey machine. As such, remove this default property and update the CI smoketest accordingly.

AlexJones0 committed:
[ot] hw/riscv: ot_earlgrey: Stop Ibex ignoring the LC state

The LC state is now sufficiently implemented for Earlgrey; this property
is now redundant.

Signed-off-by: Alex Jones <alex.jones@lowrisc.org>

AlexJones0 committed:
[ot] .github/workflows: build_test.yaml: Ignore LC on EG smoketest

In the smoketest there is no loaded OTP, so the LC state will be RAW.
This means it will not deliver a CPU_EN signal to Ibex to allow
instruction execution, which is needed for the smoketest.

Now that we do not enable this property by default, we should enable it
here in the smoketest.

Signed-off-by: Alex Jones <alex.jones@lowrisc.org>

rivos-eblot reviewed (approved):
I have not tested the EarlGrey machine with the new setting, but it seems the EG components are wired the same way as the DJ machine so it should be ok.

Could you update the `.gitlab-ci.d` CI script as well (+ a nasty omission from my side)?
````diff
diff --git a/.gitlab-ci.d/opentitan/build.yml b/.gitlab-ci.d/opentitan/build.yml
index a116e489ed..900881fe52 100644
--- a/.gitlab-ci.d/opentitan/build.yml
+++ b/.gitlab-ci.d/opentitan/build.yml
@@ -28,6 +28,7 @@ build-clang:
         sed -E 's,'"$QEMU_DIR"',@QEMU_DIR@,g' > compile_commands.json.tpl
     - ../scripts/opentitan/swexit.py -t ibexdemo -b 0x0 -o exit_id.bin
     - ../scripts/opentitan/swexit.py -t earlgrey -b 0x80 -o exit_eg.bin
+    - ../scripts/opentitan/swexit.py -t darjeeling -b 0x80 -o exit_dj.bin
   artifacts:
     public: false
     expire_in: 1 hour
diff --git a/.gitlab-ci.d/opentitan/ot-smoke.yml b/.gitlab-ci.d/opentitan/ot-smoke.yml
index 21fd36878b..f4fa1d7321 100644
--- a/.gitlab-ci.d/opentitan/ot-smoke.yml
+++ b/.gitlab-ci.d/opentitan/ot-smoke.yml
@@ -10,6 +10,7 @@ smoke-tests-ot:
         -device loader,addr=0x100080,file=build/exit_id.bin -d in_asm,int
     - timeout -s KILL 4 build/qemu-system-riscv32 -M ot-earlgrey,no_epmp_cfg=true -nographic
         -object ot-rom_img,id=rom,file=build/exit_eg.bin -d in_asm,int
+        -nographic -global ot-ibex_wrapper.lc-ignore=on
     - timeout -s KILL 4 build/qemu-system-riscv32 -M ot-darjeeling,no_epmp_cfg=true
         -object ot-rom_img,id=rom,file=build/exit_dj.bin -d in_asm,int
          -nographic -global ot-ibex_wrapper.lc-ignore=on
````

AlexJones0 committed:
[ot] .gitlab-ci.d/opentitan: ci: Update Gitlab CI

Add missing Darjeeling SW Exit to the build script, and update the
Earlgrey smoke test to make Ibex ignore the LC state as no OTP image is
loaded into the `lc_ctrl` in the smoke test run.

Co-authored-by: Emmanuel Blot <eblot@rivosinc.com>
Signed-off-by: Alex Jones <alex.jones@lowrisc.org>

rivos-eblot reviewed (approved):
LGTM. Thanks

AlexJones0 commented:
EG Regression CI failures are in the following tests:
* `//sw/device/tests/crypto:sha256_functest_sim_qemu_rom_with_fake_keys`
* `//sw/device/tests/crypto:sha384_functest_sim_qemu_rom_with_fake_keys`
* `//sw/device/tests/crypto:sha512_functest_sim_qemu_rom_with_fake_keys`

Which are known to be broken in upstream OpenTitan `earlgrey_1.0.0` at the time of the CI run. A previous run of this PR had those tests passing, so I'm ignoring those failures.
